azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message_queue.py

```diff
@@ -0,0 +1,142 @@
+import queue
+import logging
+from threading import Event
+
+logger = logging.getLogger(__name__)
+
+
+class MessageQueue(queue.Queue):
+    def __init__(self, on_enable, on_disable):
+        super(MessageQueue, self).__init__()
+        self._enabled = False
+        self._on_enable = on_enable
+        self._on_disable = on_disable
+
+    def enable(self):
+        self._enabled = True
+        self._on_enable()
+
+    def disable(self):
```

Comment:
Enable and disable have to be able to be done on a MessageQueue because it's possible to disable receiving messages for specific inputs on the module client. You can't just enable/disable the whole feature